Explorer/Assets/DCL/Backpack/BackpackSlotsController.cs

```diff
@@ -1,24 +1,69 @@
+using Cysharp.Threading.Tasks;
+using DCL.AvatarRendering.Wearables.Components;
+using DCL.Backpack.BackpackBus;
 using System;
-using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using Utility;
 
 namespace DCL.Backpack
 {
-    public class BackpackSlotsController
+    public class BackpackSlotsController : IDisposable
     {
-        private readonly Dictionary<string, AvatarSlotView> avatarSlots = new Dictionary<string, AvatarSlotView>();
+        private readonly BackpackEventBus backpackEventBus;
+        private readonly NftTypeIconSO rarityBackgrounds;
+        private readonly Dictionary<string, AvatarSlotView> avatarSlots = new ();
         private AvatarSlotView previousSlot;
 
-        public BackpackSlotsController(AvatarSlotView[] avatarSlotViews)
+        public BackpackSlotsController(AvatarSlotView[] avatarSlotViews, BackpackCommandBus backpackCommandBus, BackpackEventBus backpackEventBus, NftTypeIconSO rarityBackgrounds)
         {
+            this.backpackEventBus = backpackEventBus;
+            this.rarityBackgrounds = rarityBackgrounds;
+
+            this.backpackEventBus.EquipEvent += EquipInSlot;
+            this.backpackEventBus.UnEquipEvent += UnEquipInSlot;
+
             foreach (var avatarSlotView in avatarSlotViews)
             {
-                avatarSlots.Add(avatarSlotView.Category, avatarSlotView);
+                avatarSlots.Add(avatarSlotView.Category.ToLower(), avatarSlotView);
```

Comment:
It's not a big deal here but you could pass a comparison to the dictionary `.ctor` instead of creating a new string

Reply:
This is temporary as the .ToLower() is going to disappear as soon as I move the slots construction to a ScriptableObject